webapp/pages/settings/my-social-media.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <ds-card :header="$t('settings.social-media.name')">
+    <ds-space
+      v-if="socialMediaLinks"
+      margin-top="base"
+      margin="x-small"
+    >
+      <ds-list>
+        <ds-list-item
+          v-for="link in socialMediaLinks"
+          :key="link.url"
+        >
+          <a :href="link.url">
+            <img
+              :src="link.favicon"
+              alt="Social Media link"
+              width="16"
+              height="16"
+            >
+            {{ link.url }}
+          </a>
+        </ds-list-item>
+      </ds-list>
+    </ds-space>
+    <div>
+      <ds-input
+        v-model="value"
+        placeholder="Add social media url"
+        name="social-media"
+        :schema="{type: 'url'}"
+      />
+    </div>
+    <ds-space margin-top="base">
+      <div>
+        <ds-button
+          primary
+          @click="handleAddSocialMedia"
+        >
+          {{ $t('settings.social-media.submit') }}
+        </ds-button>
+      </div>
+    </ds-space>
+  </ds-card>
+</template>
+<script>
+import gql from 'graphql-tag'
+import { mapGetters, mapMutations } from 'vuex'
+
+export default {
+  data() {
+    return {
+      value: ''
+    }
+  },
+  computed: {
+    ...mapGetters({
+      currentUser: 'auth/user'
+    }),
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-27T13:54:02Z" title="Wednesday, March 27th 2019, 2:54:02 pm +01:00">Mar 27, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,86 @@
+<template>
+  <ds-card :header="$t('settings.social-media.name')">
+    <ds-space
+      v-if="socialMediaLinks"
+      margin-top="base"
+      margin="x-small"
+    >
+      <ds-list>
+        <ds-list-item
+          v-for="link in socialMediaLinks"
+          :key="link.url"
+        >
+          <a :href="link.url">
+            <img
+              :src="link.favicon"
+              alt=""
+            >
+            {{ link.url }}
+          </a>
+        </ds-list-item>
+      </ds-list>
+    </ds-space>
+    <div>
+      <ds-input
+        v-model="value"
+        placeholder="Add social media url"
+        name="social-media"
+        :schema="{type: 'url'}"
+      />
+    </div>
+    <ds-space margin-top="base">
+      <div>
+        <ds-button
+          primary
+          @click="handleAddSocialMedia"
+        >
+          {{ $t('settings.social-media.submit') }}
+        </ds-button>
+      </div>
+    </ds-space>
+  </ds-card>
+</template>
+<script>
+import gql from 'graphql-tag'
+import { mapGetters } from 'vuex'
+
+export default {
+  data() {
+    return {
+      value: ''
+    }
+  },
+  computed: {
+    ...mapGetters({
+      currentUser: 'auth/user'
+    }),
```


@mattwr18 you asked why the page does not get updated: You have to update `currentUser`. Import `mapMutations` put it in `methods` and use it when `apollo` successfully executes the mutation.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-03-27T13:59:00Z" title="Wednesday, March 27th 2019, 2:59:00 pm +01:00">Mar 27, 2019</time>_

---


we tried doing that, but I think we didn't push up the code to have a look if we implemented it correctly... I can add it back and push it up

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-03-28T18:51:09Z" title="Thursday, March 28th 2019, 7:51:09 pm +01:00">Mar 28, 2019</time>_

---


@aonomike and I were able to get it updating the store and now the icons and links are showing up when the user adds the social media url!! 

the favicons are quite small from twitter and facebook for example... not sure if there is a way to size them without blurring them... for the time being, I have added a 16x16 constraint on all icons, but it seems to small to me... also, we obviously will work on the other review items you brought up @roschaefer like limiting how many a user can add, validating the url in the back end, etc... 
![Screenshot at 2019-03-28 15:46:49](https://user-images.githubusercontent.com/26943915/55184590-4c041100-5171-11e9-8fa5-2d504ca278d7.png)

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-29T08:33:48Z" title="Friday, March 29th 2019, 9:33:48 am +01:00">Mar 29, 2019</time>_

---


It's OK, I think we can resize them later. For styles and visuals you can also consult @appinteractive. 

It does not look so bad to me honestly.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-29T08:33:48Z" title="Friday, March 29th 2019, 9:33:48 am +01:00">Mar 29, 2019</time>_

---


It's OK, I think we can resize them later. For styles and visuals you can also consult @appinteractive. 

It does not look so bad to me honestly.